java/core/src/main/java/com/whylogs/core/metrics/aggregators/AggregatorRegistry.java

```diff
@@ -0,0 +1,52 @@
+package com.whylogs.core.metrics.aggregators;
+
+import com.whylogs.core.metrics.components.MetricComponent;
+import lombok.Data;
+import lombok.Synchronized;
+
+import java.util.HashMap;
+import java.util.function.BiFunction;
+
+
+// Registry for aggregators. You can register a component or a type-id to an aggregator.
+// The aggregators are of IAggregator which is just a wrapper on BiFunction.
+// This allows the use of lambdas to create aggregators.
+public class AggregatorRegistry {
+    private final HashMap<String, BiFunction> namedAggregators;
+    private final HashMap<Integer, BiFunction> idAggregators;
+
+    private AggregatorRegistry() {
+        this.namedAggregators = new HashMap<>();
+        this.idAggregators = new HashMap<>();
+    }
+
+    // Instance holder to avoid double checking anti-pattern for singlton
+    private static final class InstanceHolder {
+        static final AggregatorRegistry instance = new AggregatorRegistry();
+    }
+
+    public static AggregatorRegistry getInstance() {
+        return InstanceHolder.instance;
+    }
+
+    public <T extends MetricComponent, A extends BiFunction> void register(T component,  A aggregator) {
```

Comment:
Should `A` extend `IAggrevator` ?

Reply:
I went back and forth. To do a lambda I think it needs to be able to be BiFunction

Reply:
Ah, I didn't grok that when I read the description...  It looks like we can drop `IAggregator` now?

Reply:
Ok, we could do that. There is the benefit of when we want to extend from aggregators as classes they get the merge methods. So yes it could be removed, but at the same time since IAggregtor extends BiFunction it's fine, and I think makes life a little nicer to read when making customs, *but* no not needed. Thoughts?